submissions/eoneoff/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,198 @@
+}
+
+function isInvalidPortsRange(portsRange){
+    return +portsRange[0] >= +portsRange[1] || +portsRange[0] < 1 || +portsRange[0] > 65535
+    || +portsRange[1] < 1 || +portsRange[1]>65535;
+}
+
+async function parseHost(hostName){
+    let host = '';
+    if(isIpAddress(hostName)){
+        try{
+            host = (await dns.lookup(hostName)).address;
+        } catch {
+            throw "Hostname can not be resolved";
+        }
+    } else {
+        host = hostName;
+    }
+    return host;
+}
+
+function parsePorts(ports){
+    if(isRange(ports)){
+        throw "Invalid port range argument"
+    } else {
+        let limits = ports.split("-");
+        if (isInvalidPortsRange(limits)){
+                throw "Port numbers out of range"
+            } else {
+                limits[1]++;
+                return limits;
+            }
+    }
+}
+
+async function parseArguments(args){
+    let host = "";
+    let ports = [1, 65536];
+    let hostIndex = args.indexOf('--host') + 1;
+    if(hostIndex == 0 || hostIndex == args.length){
+        throw "You must specify a host to scan ports";
+    } else {
+        host = await parseHost(args[hostIndex]);
+    }
+    let portsIndex = args.indexOf('--ports') + 1;
+    if (portsIndex != 0){
+        if(portsIndex == args.length){
+            console.log("Invalis ports specified, setting to default (1-65535)");
+        } else {
+            try{
+                ports = parsePorts(args[portsIndex]);
+            } catch {
+                console.log("Invalid ports specified, setting to default(1-65535)")
+            }
+        }
+    }
+
+    return {host, ports}
+}
+
+async function scan(host, ports, args = {silent:false,module:false}){
+    let openPorts = [];
+    let tw = twister();
+    process.stdout.write("scanning ");
+    for(let port = +ports[0]; port < ports[1]; ++port) {
+        process.stdout.write(`${tw.next().value}\b`);
+        try{
+            await new Promise((resolve, reject) =>{
+                let socket = new Socket();
+                socket.setTimeout(300);
+                socket.on('connect', () =>{
+                    process.stdout.write('. ');
+                    openPorts.push(port);
+                    socket.destroy();
+                    resolve();
+                });
+                socket.on('timeout', ()=>{
+                    socket.destroy();
+                    reject();
+                });
+                socket.on('error', () =>{
+                    socket.destroy();
+                    reject();
+                });
+                socket.connect(port, host);
+            });
+        } catch {
+
+        }
+    }
+
+    if(!args.silent || !args.module){
+        let result = '';
+        if(openPorts.length){
+            result = `Port${openPorts.length > 1 ? 's': ''} ${openPorts.join(', ')} ${openPorts.length > 1 ? 'are' : 'is'}`
+        } else{
+            result = 'No ports are';

```

Comment:
Nice use case for ternary operator here, it will allow you to use `const` not `let`

Reply:
I think that with ternary operator it will get a little too long and to complex. But if I do that, I can remove the `result` variable completely and just `console.log()` the whole expression/.